nilearn/stats/second_level_model/second_level_model.py

```diff
@@ -392,8 +403,12 @@ def fit(self, second_level_input, confounds=None, design_matrix=None):
         # Learn the mask. Assume the first level imgs have been masked.
         if not isinstance(self.mask_img, NiftiMasker):
```

Comment:
Would check_embedded_masker work here ? If it is it would avoid duplicating all this for each class and help enforce better global behavior for later.

Reply:
Although sorry it's usually used in the fit(), not there.